### PR TITLE
[Simplified login] Enhance WPComWebView for jetpack connection usage

### LINKED ISSUE #7797: [Simplified login] Enhance WPComWebViewFragment to allow using it for Jetpack Connection
For more reusability, it's better to use `WPComWebViewFragment` for this project than a custom WebView, so we need to improve the fragment with the following:
1. Use a custom Toolbar instead of the one provided by `MainActivity`.
2. Ability to use as Modal (using a cross icon instead of the back icon in the Toolbar).
3. Ability to detect multiple URLs for success instead of one.
4. Use `onBackPressedDispatcher` for back button events instead of our own implementation `BackListener`, to allow it to work with `LoginActivity`.

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #7797 

⚠️ Please don't merge before #7889 

### Description
This PR adds the following changes:
1. Converts the WPComWebView to Compose, this brings some advantages as mentioned below.
2. Allow passing multiple URLs to trigger exit from the WebView, this is needed for the Jetpack Connection flow.
3. Allow showing a cross (modal display) in the Toolbar.
4. Decouple `WPComWebView` from MainActivity: it uses its Toolbar and doesn't use the BackListener.
    This is needed if we want to launch this fragment from the LoginActivity in the future.
 
As mentioned above, the conversion to Compose brings an advantage, as the `WCWebView` component we have supports capturing back button presses, now when the user presses the back button and they have navigated to deep destinations in the WebView, we will go to the previous page instead of closing the fragment (cc @kidinov @AnirudhBhat this solves the issue mentioned [here](https://github.com/woocommerce/woocommerce-android/pull/7355#issuecomment-1239039917)).
The navigation up (using the Toolbar's navigation icon) still closes the Fragment, this aligns with how custom chrome tabs behave, and we can still customize it if needed in the Future.

### Testing instructions
Please test some scenarios that use the authenticated webview, and confirm there is no regression, some flows:

- Card reader purchase that can be accessed through the Payments screen (accessible from the More screen).
- Woo installation, this can be tested using the following steps:
    1. Uninstall Woo from one of your sites.
    2. Open the site picker.
    3. Click on "Add a store"
    4. Click on "Connect existing store".
    5. Enter the URL of the non-woo store.
    6. Click "Install Woo", and confirm the flow still works.
- Jetpack Connection: this can be tested by following the steps of #7889.

### Images/gif
For Jetpack Activation, the flow looks like this now (as modal):

https://user-images.githubusercontent.com/1657201/203388198-39d8225a-4346-43ba-8647-86b9ce5fc921.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/values/styles_base.xml [hunk 1]
```diff
@@ -786,8 +786,8 @@ theme across the entire app. Overridden versions should be added to the styles.x
     </style>
 
     <style name="Woo.Animations.Dialog">
-        <item name="android:windowEnterAnimation">@anim/slide_in_up</item>
-        <item name="android:windowExitAnimation">@anim/slide_out_down</item>
+        <item name="android:windowEnterAnimation">@anim/slide_up_fade_in</item>
+        <item name="android:windowExitAnimation">@anim/slide_down_fade_out</item>
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/values/styles_base.xml:790]: I renamed these old animations to differentiate them from the plain slide-in/out animations without fading that I added for the Jetpack connection flow.

**Comment by kidinov** [comment]
@hichamboushaba thank you! I tested the order of a card reader flow and it works as expected. The only thing I noticed, was that loading and rendering felt slower. Not sure if it's something on my end or the change related though

**Comment by hichamboushaba** [comment]
> @hichamboushaba thank you! I tested the order of a card reader flow and it works as expected. The only thing I noticed, was that loading and rendering felt slower. Not sure if it's something on my end or the change related though

Thanks for testing @kidinov, I would assume it's just because it's a Compose screen in a debuggable app, can you test on the release build and see if you notice the same slowness?

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Thanks for working on this @hichamboushaba. Code looks good and works well, tested all 3 flows using the authenticated web views and all worked as expected 👍🏼 

Ready for 🚢

**Pull request merged**